packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts

```diff
@@ -162,7 +162,12 @@ export interface ISubtreeInfo<T extends ISnapshotTree | SummaryObject> {
 export function parseSummaryForSubtrees(baseSummary: ISnapshotTree): ISubtreeInfo<ISnapshotTree> {
 	// New versions of snapshots have child nodes isolated in .channels subtree
 	const channelsSubtree = baseSummary.trees[channelsTreeName];
-	if (channelsSubtree !== undefined) {
+	// If there is ".channels" subtree or there is a groupId with no trees, then that means we use
+	// the format where we store trees inside the ".channels".
+	if (
+		channelsSubtree !== undefined ||
+		(baseSummary.groupId !== undefined && Object.keys(baseSummary.trees).length === 0)
```

Comment:
I think we should just assert that .channales sub-tree never uses groupId and move on.
In context of that specific question, I think it's fine assumption to have.

The problem with more generic logic - I do not think we can assume that presence groupId tells us anything about the emptiness of the tree. I.e. it could have groupId and be empty. In general, it's not even clear what "empty" means and why someone wants to know that without attempting to load sub-tree.

One more note that is relevant to more general case: We need to be super-clear how we encode things. I see multiple options:
1. Whole tree is always returned irrespective of groupId, only blobs are missing.
2. Sub-tree with groupID could be not returned.
3. Sub-tree with groupID could be is always not returned.

I believe we are not doing # 3. The problem with #2 - we can't answer the question if it's empty or not.
And if it's # 1, then we should not use groupId test at all in code like that.

In either # 1 or 2, we need some indication that tree/blob was not returned:
1. In # 1, presence of groupID and does not tell us if we have all the blobs or not for that sub-tree. We can test it blob by blob, but that's a bit expensive.
2. In # 2, presence of groupID and empty tree does not tell us anything yet if tree exists or not.

Reply:
I think SPO is planning to do # 1, but we should clearly reflect that in protocol, and I think add some property that says whether sub-tree blobs were included or excluded. Or agree that this is not going to happen, and have helper function that can answer that question by scanning sub-tree blobs

Reply:
It would be better if we don't have to add another prop on snapshot tree just to see whether server excluded the blobs or not when we already have one prop groupId for this.